irohad/main/iroha_conf_loader.hpp

```diff
@@ -28,6 +28,24 @@ namespace config_members {
   const char *MstSupport = "mst_enable";
 }  // namespace config_members
 
+static constexpr size_t kBadJsonPrintLength = 15;
+static constexpr size_t kBadJsonPrintOffsset = 5;
+static_assert(kBadJsonPrintOffsset <= kBadJsonPrintLength,
+              "The place of error is out of the printed string boundaries!");
+
+std::string reportJsonParsingError(const rapidjson::Document &doc,
+                                   const std::string &conf_path,
+                                   std::istream &input) {
+  const size_t error_offset = doc.GetErrorOffset();
+  const size_t print_offset =
+      std::max(error_offset, kBadJsonPrintOffsset) - kBadJsonPrintOffsset;
+  input.seekg(print_offset);
+  char json_error_buf[kBadJsonPrintLength + 1]; // includes room for trailing zero
+  json_error_buf[input.readsome(json_error_buf, sizeof(json_error_buf))] = 0;
+  return "JSON parse error [" + conf_path + "] " + "(near `" + json_error_buf
+      + "'): " + std::string(rapidjson::GetParseError_En(doc.GetParseError()));
```

Comment:
Is it intended, that `json_error_buf` is wrapped in ` and '?

Reply:
Yes, it is commonly used to distinguish an opening quote from a closing one. Pretty useful when you have plenty of them, which might be the case with JSON chunk.